test/integration_test/migration_features_test.go

```diff
@@ -330,9 +349,328 @@ func testMigrationStashStrategyWithStartApplication(t *testing.T) {
+
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "Error setting source kubeconfig")
+
+	suspendOptions := storkapi.SuspendOptions{}
+	stashStrategy := storkapi.StashStrategy{StashCR: true}
+	err = updateAppReg(appName, suspendOptions, stashStrategy)
+	require.NoError(t, err, fmt.Sprintf("Error setting stash strategy in application registrations for app %s", appName))
+
+	logrus.Infof("Starting migration %s/%s with startApplication false", appData.Ns, migNamePref+appName)
+	startApplications := false
+	mig, err := asyncdr.CreateMigration(migNamePref+appName, migrationNamespace, clusterPairName, appData.Ns, &includeVolumesFlag, &includeResourcesFlag, &startApplications)
+	err = asyncdr.WaitForMigration([]*storkapi.Migration{mig})
+	require.NoError(t, err, "Error waiting for migration")
+	logrus.Infof("Migration %s/%s completed successfully ", appData.Ns, migNamePref+appName)
+
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting dest kubeconfig")
+
+	// validate stashed cm in destination
+	stashedCMCount := getStashedConfigMapCount(t, appData.Ns)
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		require.True(t, ok, fmt.Sprintf("app %s is not currently supported in test framework", appName))
+	}
+	expectedStashedCMCount := len(operatorCRMap[appName])
+	require.Equal(t, expectedStashedCMCount, stashedCMCount, fmt.Sprintf("expected stashed configmap count %d got %d", expectedStashedCMCount, stashedCMCount))
+
+	// activate app
+	err = activateAppUsingStorkctl(appData.Ns, false)
+	require.NoError(t, err, fmt.Sprintf("Error activating app in namespace %s", appData.Ns))
+
+	err = setDestinationKubeConfig()
+	require.NoError(t, err, "Error setting dest kubeconfig")
+
+	logrus.Infof("Waiting for application pods to come up")
+	time.Sleep(5 * time.Minute)
+	logrus.Infof("Applications pods should be up by now")
+
+	// validate that CR is present
+	validated, err := validateCR(appName, appData.Ns, destClusterConfigPath)
+	require.NoError(t, err, fmt.Sprintf("Error validating CR for app %s in namespace %s in destination", appName, appData.Ns))
+	require.True(t, validated, fmt.Sprintf("CR for app %s in namespace %s should have been present in destination", appName, appData.Ns))
+
+	podsMigrated, err := core.Instance().GetPods(appData.Ns, nil)
+	require.NoError(t, err, "Error getting migrated pods")
+
+	podsMigratedLen := len(podsMigrated.Items)
+	require.Equal(t, podsCreatedLen, podsMigratedLen, "Pods migration failed as len of pods found on source doesn't match with pods found on destination")
+
+	logrus.Infof("Delete source app namespace before doing reverse migration")
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "Error setting source kubeconfig")
+
+	err = core.Instance().DeleteNamespace(appData.Ns)
+	if err != nil {
+		logrus.Errorf("Error deleting namespace %s in source: %v\n", appData.Ns, err)
+	}
+	time.Sleep(1 * time.Minute)
```

Comment:
Same thing. Maybe as an improvement we should check resource cleanup success. If for some reasons resources take time to delete test will fail while migrations are running fine.